desktop-widgets/diveshareexportdialog.cpp

```diff
@@ -105,34 +106,16 @@ void DiveShareExportDialog::finishedSlot()
 void DiveShareExportDialog::doUpload()
 {
 	//Store current settings
-	qPrefCloudStorage::set_diveshare_uid(ui->txtUID->text());
-	qPrefCloudStorage::set_diveshare_private(ui->chkPrivate->isChecked());
+	QString uid = ui->txtUID->text();
+	bool noPublic = ui->chkPrivate->isChecked();
+	qPrefCloudStorage::set_diveshare_uid(uid);
+	qPrefCloudStorage::set_diveshare_private(noPublic);
 
 	//Change UI into results mode
 	ui->frameConfigure->setVisible(false);
 	ui->frameResults->setVisible(true);
 	ui->progressBar->setVisible(true);
 	ui->progressBar->setRange(0, 0);
 
-	//generate json
-	struct membuffer buf = {};
-	export_list(&buf, NULL, exportSelected, false);
-	QByteArray json_data(buf.buffer, buf.len);
-	free_buffer(&buf);
-
-	//Request to server
-	QNetworkRequest request;
-
-	if (ui->chkPrivate->isChecked())
-		request.setUrl(QUrl(DIVESHARE_BASE_URI "/upload?private=true"));
-	else
-		request.setUrl(QUrl(DIVESHARE_BASE_URI "/upload"));
-
-	request.setRawHeader("User-Agent", getUserAgent().toUtf8());
-	if (ui->txtUID->text().length() != 0)
-		request.setRawHeader("X-UID", ui->txtUID->text().toUtf8());
-
-	reply = manager()->put(request, json_data);
-
-	QObject::connect(reply, SIGNAL(finished()), this, SLOT(finishedSlot()));
+	uploadDiveShare::instance()->doUpload(exportSelected, uid, noPublic);
```

Comment:
this follows the pattern of splitting a code move across two commits.
I understand your reason for it. maybe I shouldn't even comment on it.
this is simply a preference on my part to have that in the same commit...